### ISSUE TITLE
some roads disappeared under new version 2.9.0

### ISSUE THREAD

**Issue by linze99** [mapbox/mapbox-gl-js#12013]
**mapbox-gl-js version**:2.9.0

**browser**:Chrome

### Steps to Trigger Behavior

 1.Just use this [example](https://docs.mapbox.com/mapbox-gl-js/example/map-projection/),setting the js and css links  to version 2.9.0

### Expected Behavior
this is from version2.8.2
![282](https://user-images.githubusercontent.com/30024479/174443778-21e5509d-a431-4e6f-b9dd-9553b28fbf27.JPG)

### Actual Behavior
![290](https://user-images.githubusercontent.com/30024479/174443785-f4b50bae-b1c2-43a7-9d57-2e64bcfb3933.JPG)

**Comment by linze99** [comment]
I created a codesandbox project [here](https://codesandbox.io/s/condescending-fire-5oxc0w?file=/src/App.js)
this is the render result:
![290](https://user-images.githubusercontent.com/30024479/174468320-c07c5582-4fb4-4613-b23d-609790848c02.JPG)

but when i set the mapbox-gl version to 2.9.0-beta2,it works well
![282](https://user-images.githubusercontent.com/30024479/174468315-b483a9e4-6e97-4ef8-a3c6-b197617a39ec.JPG)

**Comment by clem4net** [comment]
Line layer seems to broken :/
I have the same problem in my website.

**Comment by codingmiao** [comment]
I have the same problem, all line layers are no longer displayed .

**Comment by mourner** [comment]
We're trying to track this down but can't reproduce it yet — @linze99 @clem4net @codingmiao can you please provide more details about your environment (OS, browser with exact versions, etc.), whether this happens in all browsers or just one, are there any warnings in the console, etc.?

**Comment by linze99** [comment]
I'm just modify the official exmaple [Display a map on a webpage](https://docs.mapbox.com/mapbox-gl-js/example/simple-map/). set the js and css version to 2.9.0
I'm using windows10,Chrome 102.0.5005.115;
this is the html code:
```js
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>Display a map on a webpage</title>
    <meta
      name="viewport"
      content="initial-scale=1,maximum-scale=1,user-scalable=no"
    />
    <link
      href="https://api.mapbox.com/mapbox-gl-js/v2.9.0/mapbox-gl.css"
      rel="stylesheet"
    />
    <script src="https://api.mapbox.com/mapbox-gl-js/v2.9.0/mapbox-gl.js"></script>
    <style>
      body {
        margin: 0;
        padding: 0;
      }
      #map {
        position: absolute;
        top: 0;
        bottom: 0;
        width: 100%;
      }
    </style>
  </head>
  <body>
    <div id="map"></div>
    <script>
      mapboxgl.accessToken =
        "pk.eyJ1IjoienBsaW56ZSIsImEiOiJjamt4azU5eHEwOWpmM3Fxa25jeGxrb2trIn0.7JolLE79Xc1c2ar6fhSbKA";
      const map = new mapboxgl.Map({
        container: "map", // container ID
        style: "mapbox://styles/mapbox/streets-v11", // style URL
        center: [-74.5, 40], // starting position [lng, lat]
        zoom: 9, // starting zoom
      });
    </script>
  </body>
</html>
```
there are two net::ERR_SSL_PROTOCOL_ERROR errors.
![222](https://user-images.githubusercontent.com/30024479/174552209-c9f75932-b3af-4703-8c74-47f13f911399.JPG)

**Comment by mourner** [comment]
@linze99 is the problem the same on e.g. Firefox?

**Comment by linze99** [comment]
I have installed the latest version of firefox, on another computer.
OS: Windows 10
Browser: Firfox 101.0.1 (64bit)
it's still the same
![firefox](https://user-images.githubusercontent.com/30024479/174587845-3072feab-8f3a-456d-a3cf-811f271ffb07.PNG)

**Comment by mourner** [comment]
Thanks! Looks like we might be hitting some kind of a GPU / graphics driver bug on Windows — working on getting it reproduced locally.

**Comment by stepankuzmin** [comment]
I'm able to reproduce this, see Google Chrome on macOS vs Microsoft Edge on Windows 11 https://jsfiddle.net/x1py9hs3/2/

![Screen Shot 2022-06-20 at 15 55 06](https://user-images.githubusercontent.com/533564/174606283-7a62a18a-73a8-4cb8-9094-586b7b8aa749.png)

![Screen Shot 2022-06-20 at 15 54 26](https://user-images.githubusercontent.com/533564/174606320-4312aa8f-de8f-4b9c-a277-eb24c2494ca7.png)

**Comment by stepankuzmin** [comment]
Looks like the the regression was introduced in debb666e5abb79db4f2807ce5bbafad88a71c359

**Comment by ansis** [comment]
We've released v2.9.1 which fixes this issue. Please upgrade!

**Comment by clem4net** [comment]
Thank you for the quick fix, it's OK for me :)
